lib/cli/commands/generate.js

```diff
@@ -31,21 +34,142 @@ addons).
   }
 
   generateBlueprint(blueprintName, argTokens) {
-    const Blueprint = require(`../blueprints/${ blueprintName }`).default;
-    let blueprint = new Blueprint();
-    let blueprintArgs = this.parseArgs.call(blueprint, argTokens.slice(1));
-    blueprint.generate(blueprintArgs);
+    const result = this.findBlueprint(blueprintName);
+
+    if (result) {
+      let blueprint = result.instance;
+      let blueprintArgs = this.parseArgs.call(blueprint, argTokens.slice(1));
+      blueprint.generate(blueprintArgs);
+    } else {
+      ui.info(`Blueprint '${ blueprintName }' doesn't exist.`);
+    }
+  }
+
+  findBlueprint(name) {
+    const split = name.split(':');
+    const hasNamespace = split.length > 1;
+    const namespace = hasNamespace ? split[0] : undefined;
+    const blueprintName = hasNamespace ? split.slice(1).join(':') : split[0];
+    const addonData = discoverAddons(process.cwd());
+    const addons = topsort(addonData).map((data) => {
+      return new Addon(data.name, data.dir);
```

Comment:
Addons can [supply their own Addon class](https://github.com/denali-js/denali/blob/master/lib/runtime/addon.js#L198) that should be used when instantiating them.